Lib/test/test_turtle.py

```diff
@@ -526,6 +546,115 @@ def test_save(self) -> None:
             with open(file_path) as f:
                 assert f.read() == "postscript"
 
+    def test_no_animation_sets_tracer_0(self):
+        s = turtle.TurtleScreen(cv=unittest.mock.MagicMock())
+
+        with s.no_animation():
+            assert s.tracer() == 0
+
+    def test_no_animation_resets_tracer_to_old_value(self):
+        s = turtle.TurtleScreen(cv=unittest.mock.MagicMock())
+
+        for tracer in [0, 1, 5]:
+            s.tracer(tracer)
+            with s.no_animation():
+                pass
+            assert s.tracer() == tracer
+
+    def test_no_animation_calls_update_at_exit(self):
+        s = turtle.TurtleScreen(cv=unittest.mock.MagicMock())
+        s.update = unittest.mock.MagicMock()
+
+        with s.no_animation():
+            s.update.assert_not_called()
+        s.update.assert_called_once()
+
+
+class TestTurtle(unittest.TestCase):
+    def test_begin_end_fill(self):
+        with patch_screen():
+            t = turtle.Turtle()
```

Comment:
You could create a turtle instance in a `setUp()` method for that:

```py
def setUp(self):
	with patch_screen():
		self.turtle = turtle.Turtle()
```

That way, you can re-use it in tests as `self.turtle`.